engine/apps/api/permissions.py

```diff
@@ -117,29 +121,24 @@ def get_most_authorized_role(permissions: LegacyAccessControlCompatiblePermissio
     return min({p.fallback_role for p in permissions}, key=lambda r: r.value)
 
 
-def user_is_authorized(
-    user: "User",
-    required_permissions: LegacyAccessControlCompatiblePermissions,
-    required_basic_role_permission: LegacyAccessControlRole = None,
-) -> bool:
+def user_has_minimum_required_basic_role(user: "User", required_basic_role: LegacyAccessControlRole) -> bool:
+    return user.role <= required_basic_role.value
+
+
+def user_is_authorized(user: "User", required_permissions: LegacyAccessControlCompatiblePermissions) -> bool:
     """
-    This function checks whether `user` has all necessary permissions. If `required_basic_role_permission` is set,
-    it only checks the basic user role, otherwise it checks whether `user` has all permissions in
-    `required_permissions`.
+    This function checks whether `user` has all necessary permissions in `required_permissions`.
     RBAC permissions are used if RBAC is enabled for the organization, otherwise the fallback basic role is checked.
-
-    user - The user to check permissions for
-    required_permissions - A list of permissions that a user must have to be considered authorized
-    required_basic_role_permission - Min basic role user must have to be considered authorized (used in cases when
-    it's needed to check ONLY the basic user role, otherwise `required_permissions` should be used)
     """
-    if required_basic_role_permission is not None:
-        return user.role <= required_basic_role_permission.value
     if user.organization.is_rbac_permissions_enabled:
         user_permissions = [u["action"] for u in user.permissions]
-        required_permission_values = [p.value for p in required_permissions]
-        return all(permission in user_permissions for permission in required_permission_values)
-    return user.role <= get_most_authorized_role(required_permissions).value
```

Comment:
this logic, plus that on line 137 (in this same function) were duplicated... moved it out into `user_has_minimum_required_basic_role`